_posts/2021-01-05-overlappable-instances.markdown

```diff
@@ -0,0 +1,263 @@
+---
+layout: post
+title:  "Overlappable instances"
+date:   2021-01-05 00:00:12 +0100
+categories: haskell effects
+---
+There are many patterns, styles and libraries which can be chosen for dealing with effects in Haskell. Today I'll try to bring closer one of them.
+
+Let's start off with a very simple program written using tagless final style.
```

Comment:
using MTL style would be more appropriate I think

Reply:
Why do you think that?

Reply:
Looks like calling this style tagless final is more appropriate, as described here:
https://serokell.io/blog/tagless-final